Sources/core/objects/Gun.cpp

```diff
@@ -0,0 +1,164 @@
+#include "Gun.h"
+#include "Tile.h"
+
+Gun::Gun(size_t id) : Item(id)
+{
+    SetSprite("icons/guns.dmi");
+    ammunition_ = 1;
+    max_ammunition_ = 1;
+}
+
+bool Gun::UseAmmo()
+{
+    if (ammunition_ <= 0)
+    {
+        ammunition_ = 0;
+        return false;
+    }
+    --ammunition_;
+    return true;
+}
+
+void Gun::ShootImpl(VDir target, const std::string& sound, const std::string& projectile_type, const std::string& casing_type)
+{
+    id_ptr_on<CubeTile> tile = GetOwner()->GetOwner();
+    id_ptr_on<IMovable> shooter = GetOwner();
+    if (tile.valid())
+    {	
+        if (UseAmmo())
+        {
+            int x = target.x;
+            int y = target.y;
+            Dir facing = shooter->GetDir();
+            if(std::abs(y) != std::abs(x))
+            {
+                shooter->Rotate(VDirToDir(target));
+                auto projectile = Create<Projectile>(projectile_type, tile->GetNeighbour(VDirToDir(target)));
+                projectile->MakeMovementPattern(target, facing);
+            }
+            else
+            {
+                if (y < 0)
+                {
+                    if (facing == 0|| facing == 1)
+                    {
```

Comment:
Do not compare `Dir` variables with pure numbers where it is possible. `if (facing == D_LEFT || facing == D_RIGHT)` is much better.